src/v/cloud_storage/cache_service.cc

```diff
@@ -451,10 +451,13 @@ ss::future<> cache::put(
       .finally([&out]() { return out.close(); });
 
     // commit write transaction
+    auto src = (dir_path / tmp_filename).native();
     auto dest = (dir_path / filename).native();
-    co_await ss::rename_file((dir_path / tmp_filename).native(), dest);
 
-    auto put_size = co_await ss::file_size(dest);
+    auto put_size = co_await ss::file_size(src);
+
+    co_await ss::rename_file(src, dest);
+    co_await ss::sync_directory(dir_path.native());
```

Comment:
The `sync_directory` is not needed for data safety (& it will add latency to the read path)

If the node crashes at this point, leaving a tmp filename, then that is safe: the tmp file will be cleaned up in clean_up_on_start next startup, and the segment can be hydrated again.